language-support/hs/bindings/src/DA/Ledger/Services/MeteringReportService.hs

```diff
@@ -128,41 +70,52 @@ timestampToIso8601 ts = ISO8601.iso8601Show ut
     st = timestampToSystemTime ts
     ut = System.systemToUTCTime st
 
-isoTimeToTimestamp :: IsoTime -> Timestamp
-isoTimeToTimestamp iso = systemTimeToTimestamp $ System.utcToSystemTime $ unIsoTime iso
-
 utcDayToTimestamp :: Day -> Timestamp
 utcDayToTimestamp day = systemTimeToTimestamp $ System.utcToSystemTime $ UTCTime day (secondsToDiffTime 0)
 
-raiseApplicationMeteringReport :: LL.ApplicationMeteringReport -> Perhaps MeteredApplication
-raiseApplicationMeteringReport (LL.ApplicationMeteringReport llApp events) = do
-  application <- raiseApplicationId llApp
-  return MeteredApplication {..}
+s2a :: Maybe S.Value -> A.Value
+s2a (Just v) = toRawAesonValue v
+s2a Nothing = A.Null
+
+a2s :: A.Value -> Maybe S.Value
+a2s v = Just $ toRawStructValue v
+
+kindToAesonValue :: S.ValueKind -> A.Value
+kindToAesonValue (S.ValueKindStructValue (S.Struct sMap)) =
+  A.Object aMap
+  where
+    aMap = KeyMap.fromMap $ fmap s2a $ Map.mapKeys Key.fromText $ Map.mapKeys TL.toStrict sMap
+
+kindToAesonValue (S.ValueKindListValue (S.ListValue list)) = A.Array $ fmap toRawAesonValue list
+kindToAesonValue (S.ValueKindStringValue text) = A.String (TL.toStrict text)
+kindToAesonValue (S.ValueKindNullValue _) = A.Null
+kindToAesonValue (S.ValueKindNumberValue num) = A.Number $ Scientific.fromFloatDigits num
+kindToAesonValue (S.ValueKindBoolValue b) = A.Bool b
+
+toRawAesonValue :: S.Value -> A.Value
```

Comment:
I believe that it would be better to use the [toJSON](https://hackage.haskell.org/package/aeson-2.1.0.0/docs/Data-Aeson.html#v:toJSON) function provided by the instance `ToJSON` defined and autogenerated in `Google.Protobuf.Struct.hs`.